tests/specs/stn/with-replay-error-mode.e2e.js

```diff
@@ -0,0 +1,152 @@
+      test: testRumRequest,
+      permanent: true,
+      body: JSON.stringify({ stn: 0, err: 1, ins: 1, spa: 1, sr: 1, loaded: 1 })
+    })
+
+    getReplayOnErrorUrl = browser.testHandle.assetURL('stn/instrumented.html', config({ session_replay: { sampleRate: 0, errorSampleRate: 1 }, session_trace: { harvestTimeSeconds: 3 } }))
+  })
+
+  function simulateErrorInBrowser () { // this is a way to throw error in WdIO / Selenium without killing the test itself
+    const errorElem = document.createElement('script')
+    errorElem.textContent = 'throw new Error("triggered! 0__0");'
+    document.body.append(errorElem)
+  }
+
+  it('switches to full mode when an error happens after page load', async () => {
+    await Promise.all([
+      browser.testHandle.expectResources(5001, true),
+      browser.url(await getReplayOnErrorUrl).then(() => browser.waitForFeatureAggregate('session_trace'))
+    ])
+    await expect(getTraceMode()).resolves.toEqual([MODE.ERROR, false])
+
+    const [resources] = await Promise.all([
+      browser.testHandle.expectResources(),
+      browser.execute(simulateErrorInBrowser)
+    ])
+    await expect(getTraceMode()).resolves.toEqual([MODE.FULL, false])
+    // The loadEventEnd is part of PT and PNT resource entry and is a node created at start of page life.
+    expect(resources.request.body.res).toEqual(expect.arrayContaining([expect.objectContaining({
+      n: 'loadEventEnd', o: 'document', t: 'timing'
+    })]))
+
+    await expect(browser.testHandle.expectResources(5002)).resolves.toBeTruthy() // double check there's nothing wrong with full mode interval harvest
+  })
+
+  it('starts in full mode when an error happens before page load', async () => {
+    let getFirstSTPayload = browser.testHandle.expectResources(5010)
+    await browser.url(await browser.testHandle.assetURL('js-error-with-error-before-page-load.html', config({ session_replay: { sampleRate: 0, errorSampleRate: 1 }, session_trace: { harvestTimeSeconds: 3 } })))
+    await browser.waitForAgentLoad()
+    await expect(getFirstSTPayload).resolves.toBeTruthy()
+    await expect(getTraceMode()).resolves.toEqual([MODE.FULL, false])
+
+    await expect(browser.testHandle.expectResources(5011)).resolves.toBeTruthy()
+  })
+
+  it.withBrowsersMatching(onlyChrome)('does not capture more than the last 30 seconds when error happens', async () => {
+    await getReplayOnErrorUrl.then(builtUrl => browser.url(builtUrl)).then(() => browser.waitForAgentLoad()).then(async () => {
+      await browser.pause(30000)
+      await Promise.all([
+        browser.testHandle.expectResources(5020), // the setup expectRes promise would've timed out already -- you will see some console errors but they don't impact validity
+        browser.execute(simulateErrorInBrowser)
+      ]).then(async ([initSTAfterErr]) => {
+        expect(initSTAfterErr.request.body.res.find(node => node.n === 'loadEventEnd')).toBeUndefined() // that node should've been tossed out by now
```

Comment:
If the window changes, this test would still work by changing the time in `browser.pause()` accordingly. The window is relative to the agent's `now()` clock, not determinable using its offset/timeOrigin.

Reply:
performance.now() (the `now()` call) is the relative timestamp to performance.timeOrigin.  But yeah, for future work we should hook into that instead of arbitrary pausing, thats a bottle neck that isn't necessary.